### PR TITLE
Replace Legacy Request

### LINKED ISSUE #29: Decaf fixes - Fix '0 items were successfully added for this event.'
## Issue Overview
When adding events to the cart using MER, it now shows

> 0 items were successfully added for this event.
https://monosnap.com/file/h4cRaZfnT8iFajjE7JizynfITOjDIu

This works if I checkout 4.10.13.p so looks like it's from the decaf changes.

## Bug report or feature request?

* [x] Bug

## Steps to Reproduce (for bugs)
Use master branch of MER and you'll need https://github.com/eventespresso/event-espresso-core/tree/BUG/rmv-debug-code for Core (unless it's been merged already).

Add a ticket to the cart and view the modal, it will show as above.

### PR DESCRIPTION
**Pull request by tn3rb**
closes #29 

Similar to work done in EE Core, this PR:

- replaces references to:
	- `EE_Request_Handler`
	- `$_REQUEST`
   with the new `Request` and `Response` classes 

- tweaks some CSS

- formats, formats, formats

### CODE REVIEW

File: EED_Multi_Event_Registration.module.php [hunk 1]
```diff
@@ -588,946 +653,1059 @@ public static function filter_ticket_selector_button_url( $button_url, \EE_Event
...
+                            'FHEE__EED_Multi_Event_Registration__display_availability_error__additional_info',
+                            sprintf(
+                                esc_html__(
+                                    '%sYour request could not be completely fulfilled due to lack of availability.%s',
+                                    'event_espresso'
+                                ),
+                                '<div class="important-notice small-text">',
+                                '</div><br />'
+                            )
+                        );
```
Comment (knazart) [inline EED_Multi_Event_Registration.module.php:1318]:
```suggestion
                    return $additional_info . apply_filters(
                        'FHEE__EED_Multi_Event_Registration__display_availability_error__additional_info',
                        sprintf(
                            esc_html__(
                                '%sYour request could not be completely fulfilled due to lack of availability.%s',
                                'event_espresso'
                            ),
                            '<div class="important-notice small-text">',
                            '</div><br />'
                        )
                    );
```

**Review by knazart — APPROVED** [review]
Looks good to me. Just have a few formatting suggestions.

File: css/multi_event_registration.css [hunk 2]
```diff
@@ -1,155 +1,216 @@
...
 
 
 #cart-results-modal-wrap-dv {
-    max-height:100% !important;
-    max-width:100% !important;
+    max-height: 100% !important;
+    max-width: 100% !important;
     text-align: center;
-	z-index: 10001
+    z-index: 10001
```
[the remark this answers is not in the document]
Comment (tn3rb) [reply css/multi_event_registration.css:155]: thnx, will fix

File: EE_Event_Cart_Line_Item_Display_Strategy.php [hunk 3]
```diff
@@ -1,412 +1,434 @@
...
-				$count = 0;
-				static $total_count = 0;
+    public function display_line_item(EE_Line_Item $line_item, $options = [])
+    {
+        $html = '';
+        // set some default options and merge with incoming
+        $default_options = [
+        'show_desc'   => true,  // 	TRUE 		FALSE
+        'event_count' => 0,
+        ];
```
[the remark this answers is not in the document]
Comment (tn3rb) [reply EE_Event_Cart_Line_Item_Display_Strategy.php:57]: oof! thnx, will fix

File: EE_Event_Cart_Line_Item_Display_Strategy.php [hunk 4]
```diff
@@ -1,412 +1,434 @@
...
+        $options         = array_merge($default_options, (array) $options);
+        $options         = (array) apply_filters(
+        'FHEE__EE_Event_Cart_Line_Item_Display_Strategy__display_line_item__options',
+        $options
+        );
```
[the remark this answers is not in the document]
Comment (tn3rb) [reply EE_Event_Cart_Line_Item_Display_Strategy.php:62]: thnx will fix

File: EE_Event_Cart_Line_Item_Display_Strategy.php [hunk 5]
```diff
@@ -1,412 +1,434 @@
...
+        EEH_HTML::strong($line_item->name()),
+        '',
+        'event-header',
+        '',
+        apply_filters(
+            'FHEE__EE_Event_Cart_Line_Item_Display_Strategy___event_row__other_attributes',
+            ' colspan="4"',
+            $line_item
+        )
+        );
```
[the remark this answers is not in the document]
Comment (tn3rb) [reply EE_Event_Cart_Line_Item_Display_Strategy.php:203]: stupid autoformat >:(

thnx will fix

File: templates/event_cart.template.php [hunk 6]
```diff
@@ -12,8 +12,8 @@
 				<thead>
 					<tr id="event-cart-tbl-row-hdr" class="event-cart-tbl-row">
 						<td class="jst-left"><?php echo __('Details', 'event_espresso');?></td>
-						<td class="jst-cntr"><?php echo __( 'Price', 'event_espresso' ); ?></td>
 						<td class="jst-cntr"><?php echo __( 'Quantity', 'event_espresso' ); ?></td>
+						<td class="jst-cntr"><?php echo __( 'Price', 'event_espresso' ); ?></td>
```
Comment (hoseinrafiei) [inline templates/event_cart.template.php:16]:
```suggestion
                                                <td class="jst-cntr"><?php echo __('Price', 'event_espresso'); ?></td>
```

File: EE_Mini_Cart_Table_Line_Item_Display_Strategy.php [hunk 7]
```diff
@@ -302,8 +306,7 @@ private function _sub_total_row( EE_Line_Item $line_item, $text = '' ) {
 				// total td
 			$html .= EEH_HTML::td( $text, '', 'total_currency total jst-rght', '', ' colspan="3"' );
 			// total td
-			$total = $line_item->total();
-			$html .= EEH_HTML::td( EEH_Template::format_currency( $total, false, false ), '', 'total jst-rght' );
+			$html .= EEH_HTML::td($line_item->total_no_code(), '', 'total jst-rght' );
```
Comment (hoseinrafiei) [inline EE_Mini_Cart_Table_Line_Item_Display_Strategy.php:309]:
```suggestion
			$html .= EEH_HTML::td($line_item->total_no_code(), '', 'total jst-rght');
```
FYI, Not important really.

**Review by hoseinrafiei — APPROVED** [review]
Looks great.

File: templates/event_cart.template.php (hunk 6 above)
Comment (tn3rb) [reply templates/event_cart.template.php:16]: thnx, will fix

File: EE_Mini_Cart_Table_Line_Item_Display_Strategy.php (hunk 7 above)
Comment (tn3rb) [reply EE_Mini_Cart_Table_Line_Item_Display_Strategy.php:309]: good eye, thnx, will fix

**Comment by Pebblo** [comment]
Unfortunately this hasn't fixed the issue with modal: https://monosnap.com/file/X3BldwHSOcANYUKQibRMTwsaPz3e6G

**Comment by tn3rb** [comment]
sry about that @Pebblo 

I had fixed it, and tested it, then somehow reverted part of the code during some other changes :\

it's working now i promise

**Comment by Pebblo** [comment]
Tested adding multiple tickets to the cart from various events.

Then tested updating the ticket QTY for some of those tickets within the cart.

Return to event list, proceed to registration, empty cart buttons all work as expected.

Mering this in!

**Pull request merged**